### PR TITLE
Remove designated array initializers in `rz_analysis.h`

### PR DESCRIPTION
**Pull request by DMaroo**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Removed designated array initializers from the code. MSVC C++ compiler seems to not be able to compile it since it is a C99 thing, and MSVC C++ does not entirely support C99.

MCVE: https://godbolt.org/z/af3v7rWfh.

Also added a new test in Appveyor for testing inclusion of Rizin headers in C++ files. The test is only enabled for VS2022.

**Test plan**

The new CI for compiling the Rizin headers included in a C++ passes succesfully, and green CI.

**Closing issues**

Fixes the build for rizinorg/cutter#3247.

<sub>(Ignore the branch name)</sub>

### CODE REVIEW

File: librz/include/rz_analysis_optype.h [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	RZ_ANALYSIS_OP_TYPE_NEW = 43,
+	RZ_ANALYSIS_OP_TYPE_ABS = 44,
+	RZ_ANALYSIS_OP_TYPE_CPL = 45, /* complement */
+	RZ_ANALYSIS_OP_TYPE_CRYPTO = 46,
+	RZ_ANALYSIS_OP_TYPE_SYNC = 47,
+// RZ_ANALYSIS_OP_TYPE_DEBUG = 43, // monitor/trace/breakpoint
+#if 0
+	RZ_ANALYSIS_OP_TYPE_PRIV = 40, /* privileged instruction */
+	RZ_ANALYSIS_OP_TYPE_FPU = 41, /* floating point stuff */
+#endif
```
Comment (wargio) [inline librz/include/rz_analysis_optype.h]: remove this?

**Review by XVilka — CHANGES_REQUESTED** [review]
I am not sure this is the right fix... 
It's better to fix the MSVC C++ compilation of the `rz_analysis.h` instead.

**Comment by XVilka** [comment]
The culprit seems to be in this code: [minimal example on Compiler Explorer](https://godbolt.org/#g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,selection:(endColumn:21,endLineNumber:5,positionColumn:21,positionLineNumber:5,selectionStartColumn:21,selectionStartLineNumber:5,startColumn:21,startLineNumber:5),source:'typedef+enum+rz_analysis_var_kind_t+%7B%0A%09RZ_ANALYSIS_VAR_KIND_INVALID+%3D+0,+///%3C+Invalid+or+unspecified+variable%0A%09RZ_ANALYSIS_VAR_KIND_FORMAL_PARAMETER,+///%3C+Variable+is+function+formal+parameter%0A%09RZ_ANALYSIS_VAR_KIND_VARIABLE,+///%3C+Variable+is+local+variable%0A%7D+RzAnalysisVarKind%3B%0A%0Astatic+const+char+*RzAnalysisVarKind_strings%5B%5D+%3D+%7B%0A%09%5BRZ_ANALYSIS_VAR_KIND_INVALID%5D+%3D+%22invalid%22,%0A%09%5BRZ_ANALYSIS_VAR_KIND_FORMAL_PARAMETER%5D+%3D+%22formal_parameter%22,%0A%09%5BRZ_ANALYSIS_VAR_KIND_VARIABLE%5D+%3D+%22variable%22,%0A%7D%3B'),l:'5',n:'0',o:'C%2B%2B+source+%231',t:'0')),k:33.333333333333336,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((h:compiler,i:(compiler:vcpp_v19_latest_x64,deviceViewOpen:'1',filters:(b:'0',binary:'1',binaryObject:'1',commentOnly:'0',debugCalls:'1',demangle:'0',directives:'0',execute:'1',intel:'0',libraryCode:'0',trim:'1'),flagsViewOpen:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,libs:!(),options:'',overrides:!(),selection:(endColumn:1,endLineNumber:1,positionColumn:1,positionLineNumber:1,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:1),l:'5',n:'0',o:'+x64+msvc+v19.latest+(Editor+%231)',t:'0')),k:33.333333333333336,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((h:output,i:(compilerName:'x86-64+gcc+13.2',editorid:1,fontScale:14,fontUsePx:'0',j:1,wrap:'1'),l:'5',n:'0',o:'Output+of+x64+msvc+v19.latest+(Compiler+%231)',t:'0')),k:33.33333333333333,l:'4',n:'0',o:'',s:0,t:'0')),l:'2',n:'0',o:'',t:'0')),version:4)

**Comment by DMaroo** [comment]
Discussion continued on Mattermost: https://im.rizin.re/rizinorg/pl/4gn8eymqupbtjnr9rahd1stzwc.

**Comment by XVilka** [comment]
Please undo the split and fix the MSVC build, preferably with adding MSVC C++ CI job too, then we can merge.

**Review by XVilka — CHANGES_REQUESTED** [review]
`clang-format` complains:
```
Formatting [1277/1629] librz/include/rz_analysis.h
librz/include/rz_analysis.h:717:26: error: code should be clang-formatted [-Wclang-format-violations]
        RZ_ANALYSIS_VAR_KIND_END  ///< Number of RzAnalysisVarKind enums
                                ^
librz/include/rz_analysis.h:734:28: error: code should be clang-formatted [-Wclang-format-violations]
        RZ_ANALYSIS_VAR_ORIGIN_END  ///< Number of RzAnalysisVarOriginKind enums
                                  ^
```

File: librz/include/rz_analysis.h [hunk 2]
```diff
@@ -728,17 +730,19 @@ typedef struct dwarf_variable_t {
...
-	[RZ_ANALYSIS_VAR_KIND_INVALID] = "invalid",
-	[RZ_ANALYSIS_VAR_KIND_FORMAL_PARAMETER] = "formal_parameter",
-	[RZ_ANALYSIS_VAR_KIND_VARIABLE] = "variable",
+static const char *RzAnalysisVarKind_strings[RZ_ANALYSIS_VAR_KIND_END] = {
+	"invalid", /* RZ_ANALYSIS_VAR_KIND_INVALID */
+	"formal_parameter", /* RZ_ANALYSIS_VAR_KIND_FORMAL_PARAMETER */
+	"variable" /* RZ_ANALYSIS_VAR_KIND_VARIABLE */
 };
 
 static const char *RzAnalysisVarOriginKind_strings[] = {
```
Comment (XVilka) [inline librz/include/rz_analysis.h]: Be consistent and use size here too, or do not use it anywhere

**Review by XVilka — CHANGES_REQUESTED** [review]

File: librz/include/rz_analysis.h (hunk 2 above)
Comment (DMaroo) [reply librz/include/rz_analysis.h]: Thanks, missed it

**Comment by DMaroo** [comment]
Will remove the draft status after the PR for adding the CI gets merged.

**Comment by XVilka** [comment]
> Will remove the draft status after the PR for adding the CI gets merged.

What PR?

**Comment by DMaroo** [comment]
I'm working on it :sweat_smile:

**Comment by XVilka** [comment]
> Will remove the draft status after the PR for adding the CI gets merged.

To be honest, I think it's better to add the CI job in the same PR since it will be red on the `dev` anyway. Just fix the syntax and add CI in this PR, it will be fine.

**Comment by DMaroo** [comment]
Alright, makes sense!

**Comment by DMaroo** [comment]
I have only enabled the C++ build test for VS2022, should I also enable it for `clang-cl`?

**Review by XVilka — APPROVED** [review]

**Pull request merged**